traits/trait_list_object.py

```diff
@@ -53,460 +94,754 @@ def __init__(self, index=0, removed=None, added=None):
+
+        if self._should_notify(removed, added):
+            self.notify(norm_index, removed, added)
+
+        return removed_item
 
     def remove(self, value):
-        trait = getattr(self, "trait", None)
-        if trait is None:
-            list.remove(self, value)
-            return
-        if trait.minlen < len(self):
-            try:
-                index = self.index(value)
-                removed = [self[index]]
-            except Exception:
-                pass
-
-            list.remove(self, value)
-
-            if self.name_items is not None:
-                self._send_trait_items_event(
-                    self.name_items, TraitListEvent(index, removed)
-                )
-        elif len(self) == 0:
-            # Let whatever system error (ValueError) should be raised be
-            # raised.
-            list.remove(self, value)
-        else:
-            self.len_error(len(self) - 1)
+        """ Remove first occurrence of value.
+
+        Parameters
+        ----------
+        value : Any
+            A value contained in the list.
+
+        Raises
+        ------
+        ValueError
+            If the value is not present.
+
+        Notes
+        -----
+        Notification:
+            index : int
+                The index between 0 and len(self).
+            added : list
+                Will be [].
+            removed : list
+                The removed item, as a list.
+
+        """
+        index = self.index(value)
+        removed = [value]
+
+        self.validate(index, removed, [])
+
+        super().remove(value)
+
+        if self._should_notify(removed, []):
+            self.notify(index, removed, [])
 
     def sort(self, key=None, reverse=False):
+        """ Sort the items in the list in ascending order, *IN PLACE*.
+        A custom key function can be supplied to customize
+        the sort order, and the reverse flag can be set to request the result
+        in descending order.
+
+        Parameters
+        ----------
+        key : Callable
+            Custom function.
+        reverse : bool
+            If true, the resulting list will be sorted in descending order.
+
+        Notes
+        -----
+        Notification:
+            index : slice
+                The slice between 0 and len(self)
+            added : list
+                A copy of the list after sorting elements.
+            removed : list
+                A copy of the list before sorting elements.
+
+        """
+
         removed = self[:]
-        list.sort(self, key=key, reverse=reverse)
-        if (
-            getattr(self, "name_items", None) is not None
-            and getattr(self, "trait", None) is not None
-        ):
-            self._send_trait_items_event(
-                self.name_items, TraitListEvent(0, removed, self[:])
-            )
+        self[:] = sorted(self, key=key, reverse=reverse)
+        added = self[:]
+        index = slice(0, len(self), None)
```

Comment:
nitpick: no need to include the optional third argument in this and other cases.